Sitka-Sound-Science-Center/sitka-landslide
# Integrate real data into main page

## Overview

Converts the main page from using the handwritten `index.json` sample file that Jeff made to using data generated by the rainfall downloader.

This didn't require many changes to the JSX at all, though I did tweak a few field names.  It also didn't require any new information, really--everything we need was already in the current and forecast data the rainfall script was getting (with the exception of weather advisories, see #16).  It mainly required a fair bit of slicing, dicing, and collating what we already had, to get it into the format needed for the frontend to display it.

In PR #7 I put the rainfall downloader in its own directory because I thought we might want to run it separately from the front-end build.  But on reflection it seems like it makes sense to just run it directly in the `getStaticProps` function in `index.js`.  So I moved it into the frontend project, which meant adding its two package dependencies as well.

Resolves #9

### Demo

![image](https://user-images.githubusercontent.com/6598836/161333593-0f7b567e-070e-4a1d-a183-dcb1bc02ded7.png)


### Notes

- The goal for the formatted rainfall data wasn't to be concise or versatile, it was to feed the frontend exactly what it needs.  So there's repetition--the "hours" list in `twentyFourHours` is a subset of the one in `threeDays`, and the info in `threeDays.days` is just a summary of what's in the `hours` list, which could have been extracted in `RiskDays.js` rather than in `rainfall.js`.  But there's no real reason for the data to be efficient or normalized, and there's significant benefit to keeping the frontend code as simple as possible, so that's what this is going for.
- The risk calculation model was calculated in millimeters of rain, and the risk calculation sticks to those units (to avoid ambiguity and also to avoid having to think about rounding and significant digits).  This adds `precipInches` and `riskPrecipInches` fields to the data, but doesn't actually use them (they show up on the detail page but not on the main page).

## Testing Instructions

- Update and run the front-end (`./scripts/update && ./scripts/server`)
- Go to http://localhost:3008/. You should see the output of the dev build in your terminal, and the site should show up looking something like the screenshot above.
- Except probably all the risks will be "Low", so go into `rainfall.js` and set `EXAGGERATION_FACTOR` to something bigger, like 10 or 15.  Reload (sometimes it recompiles automatically, but seemingly not always. not sure what the variables are) and you should see some more shape in the graph and some days in the 3-day forecast being listed as medium or high risk.
- Open the accordions.  The hours should be listed chronologically, and each should have a risk rating.  The risk rating for the day should match the highest risk for any of the hours within it.
- Break the data retrieval for each of the APIs.
   - Change the URL to make it invalid, so the request will fail outright
   - Change the URL or query params to something invalid, so the request will return an error
   You should see some details of the API error in the terminal where you ran `scripts/server`, and the front-end should show the final error: "Error: Failed to load observed or forecast rainfall data."

## Checklist
